.github/workflows/build-and-test.yml

```diff
@@ -48,6 +44,30 @@ jobs:
         run: |
           make install-tools
 
+  tidy:
+    name: tidy
+    # Use 20.04.5 until https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16450 is resolved
+    runs-on: ubuntu-20.04
+    needs: [setup-environment]
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }}
+          cache-dependency-path: '**/go.sum'
+
+      - name: tidy
+        run: |
+          make install-tools
+          make for-all CMD="make tidy"
+          if ! git diff --exit-code; then
```

Comment:
Do you know if changes are required to show the diff content in the action? It'd be helpful to do so instead of a binary check w/o info.

Reply:
This is a similar method we've been using in other jobs, e.g. https://github.com/signalfx/splunk-otel-collector/blob/main/.github/workflows/build-and-test.yml#L96 (see https://github.com/signalfx/splunk-otel-collector/actions/runs/6102809198/job/16562126153 for an example of failure). Hopefully, if there are any go.mod/go.sum changes after the tidy, the job will fail.